zephyr/lc3_decoder.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022 Bose Corporation All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIB_LIBLC3CODEC_ZEPHYR_LC3_DECODER_H_
+#define LIB_LIBLC3CODEC_ZEPHYR_LC3_DECODER_H_
+
+/* This file is a c-wrapper for the c++ LC3 lib. Please refer to the C++ API documentation
+ * for a full description of the API.
+ * General usage:
+ * 1) Call _create to allocate memory for the c++ object using the c++-lib allocation mechanism.
+ * 2) Use the wrapper functions as needed passing the allocated object as first parameter.
+ * 3) After usage call _destroy to free up the memory.
+ *    Invalid usage of the object reference after a call to _destroy cannot be detected and will
+ *    most likely result in a crash. */
+
+#include <stdint.h>
+#include "lc3_config.h"
+
+#ifdef __cplusplus
+#define LC3_EXTERN_C extern "C"
+#else
+#define LC3_EXTERN_C
+#endif
+
+typedef void* Lc3Decoder_t;
+
+#define LC3_DECODE_ERROR_FREE                       0x00
+#define LC3_DECODE_INVALID_CONFIGURATION            0x01
+#define LC3_DECODE_INVALID_BYTE_COUNT               0x02
+#define LC3_DECODE_INVALID_X_OUT_SIZE               0x03
+#define LC3_DECODE_INVALID_BITS_PER_AUDIO_SAMPLE    0x04
+#define LC3_DECODE_DECODER_ALLOCATION_ERROR         0x05
```

Comment:
Consider a consistent naming scheme for these (I assume they are error codes). 

```suggestion
#define LC3_DECODE_ERR_FREE                             0x00
#define LC3_DECODE_ERR_INVALID_CONFIGURATION            0x01
#define LC3_DECODE_ERR_INVALID_BYTE_COUNT               0x02
#define LC3_DECODE_ERR_INVALID_X_OUT_SIZE               0x03
#define LC3_DECODE_ERR_INVALID_BITS_PER_AUDIO_SAMPLE    0x04
#define LC3_DECODE_ERR_DECODER_ALLOCATION               0x05
```

Reply:
It is a copy of the values from the cpp implementation. IMO "Error free" should have been "success" or "ERR_NONE" if I follow your scheme. But I'll accept your change.